### DISCUSSION TITLE
My github page is not getting deployed.

### DISCUSSION THREAD

**Discussion by full-StackNinja** [community/community#59427] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Question

### Body

My github page is not deploying. I have previous successful multiple pages deployments. However, this time I am still getting "Your GitHub Pages site is currently being built from the main branch...". Here is the link to my project files. 
https://github.com/full-StackNinja/books-library
Please guide me.

**Comment by thunderbolt990525** [comment]
🎇 Well, I also had the issue like you.
Here are some troubleshooting steps you can follow:

Check your repository settings: Go to your repository's settings and verify that the correct branch is selected under the GitHub Pages section. Make sure you have selected "master branch" if that is the correct place of your code.

Check if your repository's GitHub Pages settings have been set to enable publishing the site: Go to your repository's settings and verify that the GitHub Pages option is enabled. If it is disabled, enable it and select the correct branch.

Wait for the build process to complete: GitHub Pages takes some time to build and publish your site. In some cases, it may take a few minutes or even hours. If your site hasn't been published yet, wait for some time and check again.

Check your code: Make sure that your code compiles and runs locally. You can do this by cloning your repository and testing your code your local machine.

Check your file names and extensions: Make sure that your files are named correctly and have the correct file extensions. GitHub Pages may not be able to build your site if your file names or extensions are incorrect.

Check your Jekyll settings: If you're using Jekyll to build your site, check your _config.yml file to verify that your settings are correct. Make sure that the destination folder matches the repository's GitHub Pages settings.

I hope one of these steps will resolve issue. Let me know if you need more assistance. ✨
